src/main/java/net/starschema/clouddb/jdbc/BQForwardOnlyResultSet.java

```diff
@@ -86,6 +86,14 @@
   private final @Nullable Long totalBytesProcessed;
   /** Whether the ResultSet came from BigQuery's cache */
   private final @Nullable Boolean cacheHit;
+  /** Specifies which mode of BI Engine acceleration was performed (if any). */
+  private final String biEngineMode;
+  /**
+   * In case of DISABLED or PARTIAL bi_engine_mode, these contain the explanatory reasons as to why
+   * BI Engine could not accelerate. In case the full query was accelerated, this field is not
+   * populated.
+   */
+  private final List<BiEngineReason> biEngineReasons;
```

Comment:
@Nullable

Reply:
For lists, it's often simpler to have them always initialized, but empty instead of null.  Do you need to distinguish null from an empty list here?

Reply:
If we were in kotlin here, I might also recommend a sealed class for BiEngineMode to capture the combination of the string and optional list of reasons, but I'm not sure of the best way to capture that in java that's not too heavy handed.

Reply:
Since it is nullable in the BQ API, I kept the same format. It also permits to remove the param in the pinger if it is null in heltool like we do with other data

Reply:
"a sealed class for BiEngineMode to capture the combination of the string and optional list of reasons" => I would do the same but since they did not in the BigQuery classes I feel they are not quite sealed in stone yet

Reply:
https://cloud.google.com/bigquery/docs/reference/rest/v2/Job